src/Compilers/Core/Portable/Symbols/Attributes/CommonTypeEarlyWellKnownAttributeData.cs

```diff
@@ -86,6 +86,8 @@ public ObsoleteAttributeData ObsoleteAttributeData
                 Debug.Assert(value != null);
                 Debug.Assert(!value.IsUninitialized);
 
+                if (_obsoleteAttributeData is { Kind: ObsoleteAttributeKind.Obsolete }) return;
```

Comment:
> if (_obsoleteAttributeData is { Kind: ObsoleteAttributeKind.Obsolete }) return; [](http://example.com/codeflow?start=15&length=80)

This change looks suspicious to me. I think the behavior should depend on order application of attributes and should match across PE and source

Reply:
> To clarify, If loading attributes from PE has some other prioritization, I think we should do the same thing for source.

Reply:
> I think the behavior should depend on order of application of attributes

In source, we currently use the order of application of attributes (last one wins).
In PE, we have a full order of precedence (Deprecated > Obsolete > WindowsExperimental > Experimental) but that isn't spec'ed. It's not clear what order of application would mean in PE, when multiple attributes are present.

As far as I know, the only order that we've spec'ed is Obsolete > Experimental, and that doesn't depend on order of application.

So I don't think we should depend on order of application. I'm okay to implement the full order of precedence in source (to match what we do in PE). Would you be okay with that?

Reply:
>  I'm okay to implement the full order of precedence in source (to match what we do in PE). Would you be okay with that?

Yes. As long as the behavior is consistent across all the "obsolete-like" attributes.